server/tests/runs.e2e-spec.ts

```diff
@@ -0,0 +1,376 @@
+        });
+
+        it('should respond with a list of runs filtered by a list of run ids', async () => {
+            const ids = user1.id + ',' + user2.id;
+            const res = await get('runs', 200, { userIDs: ids });
+            expects(res);
+            expect(res.body.totalCount).toBe(4);
+            expect(res.body.returnCount).toBe(4);
+            expect(ids).toContain(res.body.response[0].playerID.toString());
+        });
+
+        it('should respond with a list of runs filtered by flags', async () => {
+            const res = await get('runs', 200, { flags: 1 << 5 });
+            expects(res);
+            expect(res.body.totalCount).toBe(1);
+            expect(res.body.response[0].flags).toBe(run4.flags); // This uses strict equality for now, but will change in 0.10.0
+        });
+
+        it('should respond with a list of runs with the map include', () => expandTest('runs', expects, 'map', true));
+
+        it('should respond with a list of runs with the rank include', async () => {
+            const res = await get('runs', 200, { expand: 'rank' });
+            expects(res);
+            expect(res.body.totalCount).toBe(4);
+            expect(res.body.returnCount).toBe(4);
+            expect(res.body.response.filter((x) => x.hasOwnProperty('rank')).length).toBe(2); // 2 test runs have a rank, so we should see 2 in the response
+        });
+
+        it('should respond with a list of runs with the zoneStats include', () =>
+            expandTest('runs', expects, 'zoneStats', true));
+
+        it('should respond with a list of runs with the baseStats include', () =>
+            expandTest('runs', expects, 'baseStats', true));
+
+        it('should respond with a list of runs with the mapWithInfo include', async () => {
+            const res = await get('runs', 200, { expand: 'mapWithInfo' });
+            expects(res);
+            res.body.response.forEach((x) => expect(x.map).toHaveProperty('info'));
+        });
+
+        it('should respond with a list of runs filtered by partial mapName match', async () => {
+            const res = await get('runs', 200, { mapName: 'epicf', expand: 'map' });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.response[0].map.name).toBe(map2.name);
+        });
+
+        it('should respond with a list of runs that are personal bests', async () => {
+            const res = await get('runs', 200, { isPB: true, expand: 'rank' });
+            expects(res);
+            expect(res.body.totalCount).toBe(2);
+            expect(res.body.returnCount).toBe(2);
+            res.body.response.forEach(x => {
+                expect(x).toHaveProperty('rank');
+            });
+        })
+
+        it('should respond with a list of runs sorted by date', async () => {
+            const res = await get('runs', 200, { order: 'date'});
+            expects(res);
+            const sortedRes = [...res.body.response];
```

Comment:
Just a thing to keep in mind here (though it's fine in this case) - this method of object cloning is a *shallow clone*; the response objects themselves will be the same object here, the order is just different. So what you're doing is completely fine, just an important thing to understand as you learn more JS. https://developer.mozilla.org/en-US/docs/Glossary/Deep_copy for more info